client/pool.c

```diff
@@ -707,78 +829,22 @@ static int receive_data_from_connection(connection_list_element *connection)
 		conn_data->data_size = 0;
 		dfslib_uncrypt_array(g_crypt, conn_data->data, DATA_SIZE, conn_data->nfield_in++);
 
-		if(!conn_data->block_size && conn_data->data[0] == HEADER_WORD) {
-			conn_data->block = malloc(sizeof(struct xdag_block));
-
-			if(!conn_data->block) return 0;
-
-			memcpy(conn_data->block->field, conn_data->data, sizeof(struct xdag_field));
-			conn_data->block_size++;
-		} else if(conn_data->nfield_in == 1) {
-			close_connection(connection, "protocol mismatch");
+		int result = is_block_data_received(connection);
```

Comment:
Regarding the way of return value to check which function should be applied, I suggest move the handle logic in function, but leave the condition outside. It will be much more clear in receive_data_from_connection when is processing block data, when is processing share data or others.  Anyway, it's not big deal. I will try to modify this later.

Reply:
To be honest, I didn't understand your idea. Lets leave it as tech debt?